lib/Optimizer/IROptimizer/IROptimizer.cpp

```diff
@@ -30,14 +30,38 @@
 #include "llvm/Support/Casting.h"
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Support/Debug.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
 #include "llvm/Support/raw_ostream.h"
 
+#include <algorithm>
+#include <fstream>
 #include <unordered_map>
 #include <unordered_set>
 
 #define DEBUG_TYPE "ir-optimizer"
 
 namespace {
+static llvm::cl::opt<std::string> instrumentDebugDir(
```

Comment:
Nit: Looks like IROptimizer.cpp contains a lot of instrumentation and serialization logic now. I almost feel like it would be nice to move into a separate file like `IRInstrumentation.cpp`

Reply:
Done.